server/sonar-server/src/main/java/org/sonar/server/plugins/PluginCompression.java

```diff
@@ -50,12 +50,26 @@ public PluginCompression(Configuration configuration) {
     this.configuration = configuration;
   }
 
-  public void compressJar(String pluginKey, Path jarFile) {
+  public void compressJar(String pluginKey, Path sourceDir, Path targetJarFile) {
     if (configuration.getBoolean(PROPERTY_PLUGIN_COMPRESSION_ENABLE).orElse(false)) {
-      Path pack200Path = FileUtils.getPack200FilePath(jarFile);
-      pack200(jarFile, pack200Path, pluginKey);
-      String hash = calculateMd5(pack200Path);
-      RemotePluginFile compressedPlugin = new RemotePluginFile(pack200Path.getFileName().toString(), hash);
+      Path targetPack200Path = FileUtils.getPack200FilePath(targetJarFile);
+      Path sourcePack200Path = sourceDir.resolve(targetPack200Path.getFileName());
+      
+      // check if packed file was deployed alongside the jar. If that's the case, use it instead of generating it (SONAR-10395).
+      if (Files.isRegularFile(sourcePack200Path)) {
+        try {
+          LOG.debug("Found pack200: " + sourcePack200Path);
+          Files.copy(sourcePack200Path, targetPack200Path);
+        } catch (IOException e) {
+          LOG.error("Failed to copy pack200 file from " + sourcePack200Path + " to " + targetPack200Path);
+          return;
```

Comment:
I think it would be better either : 
* to pack200 the original here, instead of returning,
* to throw an ISE (as pack200 method is doing)

Because the file will be missing from compressedPlugins map, so I expect some weird issues in this case.
So throwing an exception to prevent SQ from starting maybe the best solution.